### PR TITLE
fix devicelocale, fix _checkInitLocale function

### PR DESCRIPTION
**Pull request by Overman775**
fix issue #94

### CODE REVIEW

File: lib/src/easy_localization_app.dart [hunk 1]
```diff
@@ -58,25 +58,24 @@ class _EasyLocalizationLocale extends ChangeNotifier {
...
           orElse: () => _getFallbackLocale(supportedLocales, fallbackLocale));
     }
-    //Set locale
-    if (Intl.defaultLocale == null) locale = _locale;
-    log('easy localization: Set locale ${this._locale.toString()}');
+    if (Intl.defaultLocale == null) locale = _locale;    
   }
 
-  bool _checkInitLocale(Locale locale) {
+  bool _checkInitLocale(Locale locale, Locale osLocale) {
```
Comment (aissat) [inline lib/src/easy_localization_app.dart]: @Overman775 I think can used `_osLocale` direct coz is globle var and already `init _osLocale = await _getDeviceLocale();` ,  need to remove `Locale osLocale` param

**Comment by Overman775** [comment]
Fixed 👍

**Comment by aissat** [comment]
@Overman775 when delete _osLocale from function was be same old code

**Comment by aissat** [comment]
> Fixed 👍
@Overman775 
not fixed  thinking we have an issue I tired 
```dart
String _deviceLocale = await findSystemLocale();
    print(_deviceLocale);
    print(window.locale);
    print(Intl.getCurrentLocale());
```
i changed iOS to `ar`but  always return `en`

**Comment by Overman775** [comment]
> > Fixed 👍
> > @Overman775
> > not fixed  thinking we have an issue I tired
> 
> ```dart
> String _deviceLocale = await findSystemLocale();
>     print(_deviceLocale);
>     print(window.locale);
>     print(Intl.getCurrentLocale());
> ```
> 
> i changed iOS to `ar`but always return `en`

On Android all fine, rechecked... You set up plist.info? 

Two things need to be added in plist.info

1. Localizations ( add your included languages )
2. Localized resources can be mixed - YES

https://github.com/flutter/flutter/issues/14128

**Comment by aissat** [comment]
> Fixed 👍
>         @Overman775
>         not fixed thinking we have an issue I tired
> 
>     String _deviceLocale = await findSystemLocale();
>         print(_deviceLocale);
>         print(window.locale);
>         print(Intl.getCurrentLocale());
> 
>     i changed iOS to arbut always return en
> 
> On Android all fine, rechecked... You set up plist.info?
> 
> Two things need to be added in plist.info
> 
>     Localizations ( add your included languages )
>     Localized resources can be mixed - YES
> 
> flutter/flutter#14128

yeah on android everything working well

**Comment by Overman775** [comment]
> yeah on android everything working well

I’ll check on iOS tomorrow, I think the problem is in the settings project

**Comment by aissat** [comment]
;

> yeah on android everything working well
> 
> I’ll check on iOS tomorrow, I think the problem is in the settings project
nice
@Overman775  plz if u can check this #78 #93 everything working well for me

**Comment by Overman775** [comment]
https://flutter.dev/docs/development/accessibility-and-localization/internationalization#appendix-updating-the-ios-app-bundle

**Comment by Overman775** [comment]
> ;
> 
> > yeah on android everything working well
> > I’ll check on iOS tomorrow, I think the problem is in the settings project
> > nice
> > @Overman775  plz if u can check this #78 #93 everything working well for me

Ok

**Comment by aissat** [comment]
@Overman775 check #97

**Comment by Overman775** [comment]
Work fine, just add locale and set Yes in project(upper message)... but new bug

```
[log] easy localization: Device locale ru_US ---Bug---Country Code 
[log] easy localization: Set locale ru
[MyApp# locale] ru
[MyApp# Intl.defaultLocale] ru
[_MyHomePageState#32c97] Привет!
```

Снимок экрана 2020-03-21 в 01 16 11
Снимок экрана 2020-03-21 в 01 16 44
Снимок экрана 2020-03-21 в 01 17 13

**Comment by Overman775** [comment]
and ui.window.locale.countryCode return US on iOS...Magic :)

**Comment by Overman775** [comment]
> and ui.window.locale.countryCode return US on iOS...Magic :)

its not a bug.. just need select region to in iOS settings

```
[log] easy localization: Device locale ru_RU
[log] easy localization: Set locale ru
[MyApp# locale] ru
[MyApp# Intl.defaultLocale] ru
[_MyHomePageState#ea2e1] Привет!
```

**Comment by Overman775** [comment]
@aissat I think I need to add instructions on how to set up a project (add a locale, mix locales -YES) in iOS

**Comment by Overman775** [comment]
reconfigured iOS project in example

**Comment by aissat** [comment]
@Overman775 thnx Man great work , just need to update ur develop Brach coz yesterday I pushed new update #97 then we need working on Readme I mean update documentation

**Comment by Overman775** [comment]
All done 😃

**Comment by Overman775** [comment]
if find problem  #78... notifyListeners and async conflict

**Comment by Overman775** [comment]
FutureBuilder fix is ​​important, it was rebuilt all the time.
I think issue #78 #93 was due to this bug

**Comment by aissat** [comment]
@Overman775 done thnx

**Comment by Overman775** [comment]
I have a small optimization idea, I'll try tomorrow 😃

**Pull request merged**